src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/LimitedUseItem.java

```diff
@@ -0,0 +1,117 @@
+package io.github.thebusybiscuit.slimefun4.implementation.items;
+
+import java.util.List;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.bukkit.NamespacedKey;
+import org.bukkit.Sound;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.persistence.PersistentDataContainer;
+import org.bukkit.persistence.PersistentDataType;
+
+import io.github.thebusybiscuit.cscorelib2.chat.ChatColors;
+import io.github.thebusybiscuit.slimefun4.api.SlimefunAddon;
+import io.github.thebusybiscuit.slimefun4.core.handlers.ItemUseHandler;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import io.github.thebusybiscuit.slimefun4.implementation.items.magical.staves.StormStaff;
+import io.github.thebusybiscuit.slimefun4.utils.LoreBuilder;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Objects.Category;
+import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
+
+/**
+ * This class represents an item with a limited number of uses.
+ * When the item runs out of "uses", it breaks.
+ *
+ * @author Linox
+ * @author Walshy
+ * @author TheBusyBiscuit
+ * @author martinbrom
+ *
+ * @see StormStaff
+ */
+public abstract class LimitedUseItem extends SimpleSlimefunItem<ItemUseHandler> {
+
+    @ParametersAreNonnullByDefault
+    public LimitedUseItem(Category category, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
+        super(category, item, recipeType, recipe, null);
+
+        addItemHandler(getItemHandler());
+    }
+
+    /**
+     * Returns the number of times this item can be used.
+     * Must be greater than zero.
+     *
+     * @return The number of times this item can be used.
+     */
+    protected abstract int getMaxUseCount();
```

Comment:
Really no point in making this method abstract, especially because it would allow you to return anything, such as `-1`, `Infinity` or `NaN`.

It would be better to control this via a setter and getter so we can validate parameters.

Reply:
Well, I did it the same way as `AContainer` does...the item is not registered if the getter returns an invalid value

Reply:
`AContainer` uses setters and getters though, like I suggested...

Also `AContainer` is not the go-to example for best practices :LUL: